src/Nervos/CellInput.h

```diff
@@ -0,0 +1,38 @@
+// Copyright © 2017-2022 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#pragma once
+
+#include "OutPoint.h"
+#include <nlohmann/json.hpp>
+
+using json = nlohmann::json;
+
+namespace TW::Nervos {
+
+/// Nervos cell input.
+struct CellInput {
+    /// Reference to the previous transaction's output.
+    OutPoint previousOutput;
+
+    /// Prevents the transaction to be mined before an absolute or relative time.
+    uint64_t since;
+
+    /// Initializes a cell input with a previous output and since
+    CellInput(OutPoint previousOutput, uint64_t since)
+        : previousOutput(std::move(previousOutput)), since(since) {}
+
+    /// Encodes the transaction into the provided buffer.
+    void encode(Data& data) const;
+
+    /// Encodes the output into JSON format.
+    json JSON() const;
```

Comment:
let's use lowercase `json` as property here, no need to use `using json = nlohmann::json;` if only one place
```suggestion
    nlohmann::json json() const;
```

Reply:
Done